Source/WebKit/NetworkProcess/cocoa/NetworkDataTaskCocoa.mm

```diff
@@ -387,6 +387,15 @@ static float toNSURLSessionTaskPriority(WebCore::ResourceLoadPriority priority)
 
 void NetworkDataTaskCocoa::didCompleteWithError(const WebCore::ResourceError& error, const WebCore::NetworkLoadMetrics& networkLoadMetrics)
 {
+    if (networkLoadMetrics.responseBodyBytesReceived > totalBytesTransferredOverNetwork()) {
+        WTFEmitSignpost(m_task.get(), DataTask, "received 0 bytes");
+
+        setTotalBytesTransferredOverNetwork(networkLoadMetrics.responseBodyBytesReceived);
+
+        if (m_client)
+            m_client->didReceiveData(WebCore::SharedBuffer::create());
+    }
+
```

Comment:
Adding this workaround for older network stack, the test was happy with Ventura machines. But this workaround postpone the network usage metrics monitoring of ResourceMonitor until the end of the communication. If the resource is much bigger than threshold, unloading won't happen until the completion.

Reply:
Presumably this is because we folded the `bytesTransferredOverNetwork` (which is actually a delta) argument in to the `DidReceiveData` IPC.

I think a cleaner refactoring might be to move all of the network resource delta computation logic out of NetworkProcess and in to WebProcess, e.g. by:

- Changing `WebResourceLoader::DidReceiveData` IPC to report the total bytes transferred over the network over this load so far rather than a delta.
- When processing the `WebResourceLoader::DidFinishResourceLoad` (which already gets a `NetworkLoadMetrics` object), do any final delta computations there and report that final delta to the resource monitor.

Probably does not need to block this patch but I think things are probably easier to reason about this way...

Reply:
No that doesn't work because we need to distinguish cached resource, blobs and data urls which won't consume network usage. We need to track this in Network Process.

Reply:
Ah, sorry, you are saying using `NetworkLoadMetrics`. But that prevents measuring during the communication which is new stack is supporting.

Reply:
Okay, I agree with your idea. I'll do that in the following patch. That way is better way for other platforms.